### ISSUE TITLE
Improve support for Firefox Mobile

### ISSUE THREAD

**Issue by momothereal** [sindresorhus/refined-github#1496]
There are a few issues when using the add-on on the Firefox Mobile browser.

I think it would be useful to improve the mobile site in general.

**General**
  - [ ] Add mobile detection (#1497)
  - [ ] Disable hotkey features on mobile

**User Profile**
  - [ ] Gists are not linked (#1497)
  - [ ] "Followers you know" isn't shown

**User Issues / PRs**
  - [ ] "Yours" and "Commented" tabs are not displayed (`add-yours-menu-item` & `add-commented-menu-item`)

**Repository**
  - [ ] ['More' repository navigation items don't display correctly](https://user-images.githubusercontent.com/6775216/44743245-f9c21b00-aacf-11e8-9bc4-b4818b68b6ff.jpg)
  - [ ] Reaction avatars are not shown
  - [ ] CI check isn't shown
  - [ ] Default branch link isn't shown
  - [ ] "Compare" and "Dependencies" pages show the desktop site¹
  - [ ] Comment "time machine" isn't shown
  - [ ] "Copy Path" button isn't shown
  - [ ] No way of viewing raw markdown file
  - [ ] "Original Poster" label isn't shown (`op-label`)
  - [ ] "Projects" tab isn't hidden when there are no projects in a repo (`remove-projects-tab`)
  - [ ] No way of uploading attachments to issue (`upload-button`)
  - [ ] Issue close cause (`Closed in ...`) isn't shown

**Code Diff**
- [ ] Cancelling inline comments does not show warning (`add-confirmation-to-comment-cancellation`)
- [ ] Options for Unified/Split and "No whitespace" are not shown

-----------------

1. This appears to be the default GitHub behavior, because these pages are not normally accessible on the mobile site. If they are kept linked in Refined Github, they should be made mobile friendly.

**Comment by hkdobrev** [comment]
Wow, I didn't even know Firefox mobile allows add-ons. I wish it was true for Chrome as well. There are so many things which could be fixed on mobile with just front-end changes. E.g. all the notifications enhancements we've made on desktop would make even more sense on mobile. That's why I think fixing mobile issues with the extension make great sense. I've created the mobile label for these issues.

**Comment by momothereal** [comment]
I think it would be a good idea to go through all the features and test with the mobile site. I'm guessing some CSS selectors are slightly different on mobile.

**Comment by hkdobrev** [comment]
@momothereal OK, let's do things one-by-one. We can keep this issue about general Firefox mobile support and it'd be wonderful if you could update the description with a checklist of things found as inconsistencies and we can create separate issues for bigger discussions on a specific feature if needed and separate PRs for individual fixes.

Could you make a checklist with task items? https://help.github.com/articles/about-task-lists/

**Comment by momothereal** [comment]
Will do!

**Comment by momothereal** [comment]
I gathered a list of things in a few minutes (see OP). I'll add more items if I find anything else.

**Comment by sindresorhus** [comment]
Is this about the "mobile" version of github.com or just desktop github.com on a mobile browser? If it's the former, I'm a huge 👎 to this. I'm not interested in supporting to completely different sites, and it also looks like GitHub is in the process of gradually making the desktop version responsive, so it seems the mobile version is slowly on its way out.

**Comment by hkdobrev** [comment]
@sindresorhus My view is this is about mobile version of GitHub on a mobile browser. But why do you say the mobile version is on its way out? What I mean by mobile version is just the responsive version of GitHub.

**Comment by bfred-it** [comment]
As @sindresorhus I don't think Refined GitHub is gonna support mobile. It's basically a different site and it's not worth the effort.

**Comment by momothereal** [comment]
To clarify, I am talking about the mobile site that is only shown when the UA is for a mobile device.

My point of view is that there is no harm in at least fixing the [visual regressions](https://user-images.githubusercontent.com/6775216/44743245-f9c21b00-aacf-11e8-9bc4-b4818b68b6ff.jpg) caused by the extension (which is the reason I opened the issue originally).

Most of the issues in the list can be fixed by simply changing a few CSS selectors when on the mobile device.

> I'm not interested in supporting to completely different sites
> It's basically a different site

I disagree. It's the same API, and most of the information needed for the different features are also present on the mobile pages. The DOM structure is relatively the same, with a few tweaks in CSS classes. However, I was not implying that *all* features should or can be ported to the mobile site. [Some features are actually really easy to port](https://github.com/sindresorhus/refined-github/pull/1497/files). Some aren't, and that's fair.

Again,
> I think it would be useful to improve the mobile site in general.

I think the mobile site could use some love, because it's terrible at doing anything but quickly looking at conversations.

If improving QOL of mobile users goes against the vision of this extension, then I'm sorry for wasting your time 😄

**Comment by sindresorhus** [comment]
I'm sorry but the mobile site is out of scope for Refined GitHub. We already have way too much to maintain with the desktop site.

**Comment by Turbo87** [comment]
hmm, I understand the point about not wanting to maintain the extension for two different sites in parallel, but the problem right now is that the desktop changes also get applied to the mobile page. this seems like a bug to me because the changes that this extension applies should be scoped to just the desktop page, and not affect the mobile view at all.

**Comment by yakov116** [comment]
> I'm sorry but the mobile site is out of scope for Refined GitHub. We already have way too much to maintain with the desktop site.

@sindresorhus now that IOS is being supported does this issue get to be reopened?

**Comment by sindresorhus** [comment]
This issue is unrelated. This was about supporting the separate (now defunct) mobile web version of GitHub. What we have today on mobile is just the desktop version optimized for mobile. I suggest opening a new issue.

**Comment by fregante** [comment]
Generally I've been trying to ensure features don't break in the mobile view… if I remember, but often they still need custom implementations or we just have to deal with the fact that some features will not work on mobile. For example GitHub does not let you delete branches on mobile, so rename/clone is also out of the question.
